numpy/lib/tests/test_function_base.py

```diff
@@ -1407,6 +1447,52 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    def test_positional_regression_9477(self):
+        # This supplies the first keyword argument as a positional,
+        # to ensure that they are still properly forwarded after the
+        # enhancement for #9477
+        f = vectorize((lambda x: x), ['float64'])
+        r = f([2])
+        assert_equal(r.dtype, np.dtype('float64'))
+
+    def test_subclass_regression_9477(self):
+        # Test that subclasses written prior to #9477 still work;
```

Comment:
This is a bit of a megatest, simply because there's so many things that a subclass could do which could potentially break in orthogonal ways when the superclass is modified.  You know, fragile base class problem and all that.  Even then, I still doubt this test is anywhere near comprehensive.